src/s11/handlers/create_session_handler.c

```diff
@@ -146,7 +166,15 @@ create_session_processing()
 
 	gtpv2c_ie ies[15];
 
-	bswap8_array(g_csReqInfo->IMSI, imsi, BINARY_IMSI_LEN);
+	// check imsi odd or even, then handle it differently
+	if ( (g_csReqInfo->IMSI[0] & 0x08) != 0  ){
```

Comment:
This check is checking the 0th byte of IMSI. This is not correct.  
As per spec, the 0th byte of EPS mobile Identity contains the Odd/Even Indication.

Reply:
"As per spec, the 0th byte of EPS mobile Identity contains the Odd/Even Indication.", correct. Through a pcap analysis to both s11 and s1ap, mme takes “odd/even indication & type of identity” (4bits) from s1ap interface as part of imsi. This is why this patch use part of the imsi to check the odd or even.

I also showed the pcap to Badhrinathpa how this happens.

Reply:
Are we checking against '9' that is received in IMSI as you have shown from capture?
I think this needs to be handled more on s1ap side as well where imsi is received and parsed. There is a todo for the same issue with IMSI at the following location:
src/s1ap/handlers/s1ap_msg_delegator.c : check comment and code on line 212-219

Reply:
This patch only fix the issue in S11. The 9 or “odd/even indication & type of identity” is useful after receive it from s1ap and we should store it in mme. But whether to store it inside imsi or outside, we can review it again and design it. Then we can start look at it from S1AP, and it should be several places in the code need change.